src/btree/bt_cursor.c

```diff
@@ -331,6 +331,17 @@ __wt_cursor_valid(WT_CURSOR_BTREE *cbt, WT_UPDATE **updp, bool *valid)
         cell = WT_COL_PTR(page, cip);
         if (__wt_cell_type(cell) == WT_CELL_DEL)
             return (0);
+        /* Check for an update. */
+        WT_RET(__wt_txn_read(session, cbt, cbt->ins == NULL ? NULL : cbt->ins->upd, &upd));
+        if (upd != NULL) {
+            if (upd->type == WT_UPDATE_TOMBSTONE) {
+                if (F_ISSET(upd, WT_UPDATE_RESTORED_FROM_DISK))
+                    __wt_free_update_list(session, &upd);
+                return (0);
+            }
+            if (updp != NULL)
+                *updp = upd;
+        }
```

Comment:
There's a large comment at the start of this function, that looks like it needs updating? Not sure, to be clear, but maybe worth checking.

Reply:
I'd rather defer this until a bit later since every function that calls `__wt_txn_read` is going to have undergo a significant refactor at some point. The comment has been wrong for a long time now since we began history store searching within `__wt_txn_read` in relevant history.